### ISSUE TITLE
Create a NuGet package

### ISSUE THREAD

**Issue by stakx** [imazen/lightresize#11]
### NuGet package requirements: ###

- It should multi-target both .NET Framework versions 2 and 4 (since these run on different versions of the CLR and we don't want to cause version conflicts for `System.*` assembly references).

### Code changes: ###

- [X] Check whether the new MSBuild-based project file format for .NET Core supports the `net20` target. If so:
- [X] Update the project files to the new .NET Core / .NET Standard format (MSBuild-based, but simplified). This makes it relatively easy to create a NuGet package for several targets..
- [x] Having both `SinglePurposeResize.cs` and `ResizeJob.cs` in a NuGet package would not make much sense. The original purpose of these two files were for demonstration's sake. Keep those files accessible via the `README`, but don't use them as is for the NuGet package.
- [x] For the same reason, having everything in only two code files and mentioning LoC limits no longer has the same relevance as before. Therefore, put each type in a separate file (per the usual .NET conventions).
- [x] Also, reformat the code according to usual .NET formatting conventions. Use StyleCop to steer this process.
- [x] Clean up the public API. Once published, it should remain relatively stable. Make the API similar to ImageResizer's (to enable an easy migration path) but not at the cost of making it too complex. LightResize is supposed to be "light", after all.
- [x] Add validation for public properties and parameters.
- [x] <strike>After all the above changes, bump the version straight up to 2.0.0, since LightResize will have a very different API than it had originally.</strike>

### Documentation changes: ###

- [x] Consider rewriting the `README` page to show some concrete usage examples. <strike>Move the current notes about LightResize's origin to wiki pages.</strike> (Good idea, but can be done later.)
- [x] Add NuGet badge to `README`.

**Comment by stakx** [comment]
There's quite a bit more that should happen before a NuGet package gets published:

- [ ] Having both `SinglePurposeResize.cs` and `ResizeJob.cs` in a NuGet package would not make much sense. The original purpose of these two files were for demonstration's sake. Preserve those files in the `README`, but don't have them as is in the NuGet package.
- [ ] For the same reason, having everything in only two code files and having LoC limits is no longer relevant. Therefore, put each type in a separate file (per the usual .NET conventions).
- [ ] Also, reformat the code according to usual .NET formatting conventions. Use StyleCop to steer this process.
- [ ] Clean up the public API. Once the NuGet package is out there, the API shouldn't be changed too much anymore. The API should be as similar to ImageResizer's as possible without becoming bloated.
- [ ] After all the above changes, bump the version straight up to 2.0.0, since LightResize will have a very different API than it had originally.
